core/src/library.rs

```diff
@@ -158,6 +158,15 @@ impl<'gc> MovieLibrary<'gc> {
     /// Registers an export name for a given character ID.
     /// This character will then be instantiable from AVM1.
     pub fn register_export(&mut self, id: CharacterId, export_name: AvmString<'gc>) {
+        let character_exists = self.contains_character(id);
+        debug_assert!(character_exists);
+        if !character_exists {
+            tracing::error!(
+                "Tried to register export '{export_name}' for a non-existent character {id}"
+            );
+            return;
+        }
```

Comment:
There's nothing preventing a malformed SWF from exporting a character before defining it, so this probably shouldn't be an assert. I suspect such an export will work correctly in FP, but this would need testing.

Reply:
I'm not sure how that works in Ruffle or FP, but it's currently impossible in Ruffle. The method `register_export` is called only when the character is registered.